graylog2-server/src/main/java/org/graylog2/featureflag/ImmutableFeatureFlags.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog2.featureflag;
+
+import com.codahale.metrics.Counter;
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.MetricRegistry;
+import org.graylog2.shared.metrics.MetricUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static org.graylog2.featureflag.FeatureFlagStringUtil.stringFormat;
+import static org.graylog2.featureflag.FeatureFlagStringUtil.toUpperCase;
+
+class ImmutableFeatureFlags implements FeatureFlags {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ImmutableFeatureFlags.class);
+    private final Map<String, FeatureFlag> flags;
+
+    public ImmutableFeatureFlags(Map<String, String> flags, MetricRegistry metricRegistry) {
+        this.flags = flags.entrySet().stream()
+                .collect(Collectors.toMap(e -> toUpperCase(e.getKey()), e -> new FeatureFlag(e.getKey(), e.getValue(), metricRegistry)));
+    }
+
+    @Override
+    public Map<String, String> getAll() {
+        return flags.values().stream().collect(Collectors.toMap(flag -> flag.name, flag -> flag.value));
+    }
+
+    @Override
+    public boolean isOn(String feature, boolean defaultValue) {
```

Comment:
I was wondering what the use case for the `defaultValue` is. According to the pitch a feature flag should be off by default, when would one set the `defaultValue` to true?

Reply:
The purpose of the defaultValue was to give the developer some flexibility and to make clear what the result would be if the feature is not present. But you are right I can't think of any use case where I would set the default value to true. Maybe it's not worth it to have this flexibility. If we remove the default parameter we should provide at least some JavaDoc to make clear what is the fallback behavior or come up with good name for the method.

Reply:
Sounds good! We can always add a method with a default value later if we see that we need it. Taking the parameter away is way harder once the API is in use. :+1:

Reply:
I removed the defaultValue parameter.